### PR TITLE
[Fix] Image padding / cropping validation

### PR DESCRIPTION
**Pull request by ChiragSW**
Fixes part 1 of #22595 

## Contributor Agreement
**Please check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

> **Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

**Comment by ChiragSW** [comment]
@hertschuh please review

File: keras/src/ops/image.py [hunk 1]
```diff
@@ -1047,6 +1047,45 @@ def call(self, images):
...
+                raise ValueError(
+                    f"{name} must be >= 0. Received: {name}={value}"
+                )
+
+        validate_non_negative(self.top_padding, "top_padding")
+        validate_non_negative(self.bottom_padding, "bottom_padding")
+        validate_non_negative(self.target_height, "target_height")
+        validate_non_negative(self.left_padding, "left_padding")
+        validate_non_negative(self.right_padding, "right_padding")
+        validate_non_negative(self.target_width, "target_width")
```
Comment (hertschuh) [inline keras/src/ops/image.py]: `compute_output_spec` is often not called. But this part of the validation should be done in the op itself here:
https://github.com/keras-team/keras/blob/master/keras/src/ops/image.py#L1121

File: keras/src/ops/image.py [hunk 2]
```diff
@@ -1260,6 +1345,45 @@ def call(self, images):
...
+                raise ValueError(
+                    f"{name} must be >= 0. Received: {name}={value}"
+                )
+
+        validate_non_negative(self.top_cropping, "top_cropping")
+        validate_non_negative(self.bottom_cropping, "bottom_cropping")
+        validate_non_negative(self.target_height, "target_height")
+        validate_non_negative(self.left_cropping, "left_cropping")
+        validate_non_negative(self.right_cropping, "right_cropping")
+        validate_non_negative(self.target_width, "target_width")
```
Comment (hertschuh) [inline keras/src/ops/image.py]: Same for this one.

Also create a common `validate_non_negative` for both.

File: keras/src/ops/image_test.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import tensorflow as tf
 from absl.testing import parameterized
 
+import keras
```
Comment (hertschuh) [inline keras/src/ops/image_test.py]: Do not `import keras` in unit test. Only import specific things from `keras.src`.

File: keras/src/ops/image_test.py [hunk 4]
```diff
@@ -221,6 +222,34 @@ def test_crop_images(self):
         out = kimage.crop_images(x, 2, 3, target_height=10, target_width=20)
         self.assertEqual(out.shape, (3, 10, 20))
 
+    def test_pad_images_invalid_rank_keras_input(self):
+        x = keras.Input(shape=(16,))
```
Comment (hertschuh) [inline keras/src/ops/image_test.py]: Use a `KerasTensor` like you did for the other test. This way you don't need `keras.Input`.

Same for all the other tests below.

File: keras/src/ops/image_test.py [hunk 5]
```diff
@@ -2609,6 +2637,90 @@ def test_crop_images_unknown_shape(self):
...
+            )
+        with self.assertRaisesRegex(ValueError, "target_height must be >= 0"):
+            kimage.PadImages(
+                top_padding=0,
+                left_padding=0,
+                bottom_padding=None,
+                right_padding=None,
+                target_height=-1,
+                target_width=10,
+            ).symbolic_call(x)
```
Comment (hertschuh) [inline keras/src/ops/image_test.py]: Use `kimage.pad_images` instead of `PadImages(...).symbolic_call()`.

File: keras/src/ops/image_test.py [hunk 6]
```diff
@@ -2609,6 +2637,90 @@ def test_crop_images_unknown_shape(self):
...
+        with self.assertRaisesRegex(ValueError, "target_height must be >= 0"):
+            kimage.PadImages(
+                top_padding=0,
+                left_padding=0,
+                bottom_padding=None,
+                right_padding=None,
+                target_height=-1,
+                target_width=5,
+            ).symbolic_call(x)
```
Comment (hertschuh) [inline keras/src/ops/image_test.py]: Use `kimage.pad_images` instead of `PadImages(...).symbolic_call()`.

File: keras/src/ops/image_test.py [hunk 7]
```diff
@@ -2609,6 +2637,90 @@ def test_crop_images_unknown_shape(self):
...
+        with self.assertRaisesRegex(ValueError, "target_height must be >= 0"):
+            kimage.CropImages(
+                top_cropping=0,
+                left_cropping=0,
+                bottom_cropping=None,
+                right_cropping=None,
+                target_height=-1,
+                target_width=5,
+            ).symbolic_call(x)
```
Comment (hertschuh) [inline keras/src/ops/image_test.py]: Use `kimage.crop_images` instead of `CropImages(...).symbolic_call()`.

**Review by hertschuh — COMMENTED** [review]
Thanks!

**Comment by ChiragSW** [comment]
The requested changes are applied.

**Review by hertschuh — APPROVED** [review]
Thanks!
